### ISSUE TITLE
[REQUEST] Easier Retype

### ISSUE THREAD

**Issue by j1my3p1x** [elwyos/nintype#5]
Main idea: A shortcut action. 
You activate it once to select and activate retype for the last typed word. Press it again and it leaves the last type word as is and moves to the second to last typed word. And so on and so forth. 

It would be so convenient. Any way to quickly retype words  without your fingers having to leave the keyboard keys would just be insanely nice. Maybe an option replacing the bird thing that shows what you typed with a thing that replaces the word currently being typed. It would just be so sweet. 

Thanks, and sorry for suggesting so much. I hope I’m not stressing you out or anything.

**Comment by elwyos** [comment]
No, these are excellent, at least now I have a proper repository to look at suggestions. I can do this as a shortcut, I'll put this one on the todo list, should be easy to get done, maybe even by next beta (hopefully tonight or tomorrow)

**Comment by elwyos** [comment]
I've added a retype shortcut, RT0 (left), RT1 (mid) and RT2 (right), which would correspond to the words on the topbar.

For example, I've bound force press return button to RT1, so if I want to retype the word on the middle, I just force press the enter key. I think it can work alright, if only there are more keys to access though

**Comment by j1my3p1x** [comment]
That's sweet. Thanks!

**Comment by j1my3p1x** [comment]
Well I’ve tried to come up with a more effective idea. 

Retype Seamless - It enters retype mode for the word on the very right of the topbar, (just like the suggestion below too) and if you activate it again it will select the the word in the middle of the topbar. If you press it again it’s the word on the left, if you press it again then it’s the word behind the word on the left of the topbar that gets put in retype mode. 

Retype Erase - Everytime you activate it it deletes the word currently in or behind the cursor and then automatically enters retype mode for the word behind the deleted word. If a word is in retype mode and you press this key again then it deletes the current word and enters retype mode for the word behind it.

You can activate the gesture as many times as needed to continue writing your sentence, which would be really fast, especially if it took you writing ten more words after an error for you to notice a mistake.

**Comment by elwyos** [comment]
Hmm, the problem with retype seamless is that iOS doesn't currently have a
"highlight these words" for retyping function, but I suppose I can make do
with a "retyping" indicator, but we'll have to see.

There's currently no such thing as "retype mode" for a word, there's only
"retype" action, which deletes a word and places the cursor in its place.

It's not a bad mode to have though, it could be rather useful. I'll have to
think about this for a bit though

**Comment by j1my3p1x** [comment]
Oh, you must be doing something different on Android then. I've only ever used the Android version. Thanks for getting back. Maybe adding a \* in front of the word as an indicator could work to show selection.
